models/bucket.py

```diff
@@ -33,6 +36,13 @@ def __init__(self, **attributes):
         self.current_amount = attributes.get('current_amount', 0)
         self.comment = attributes.get('comment', None)
         self.complete = False
+        
+        #create a list of accepted frequencies
+        valid_frequencies = {None, "weekly", "2_weeks", "monthly", "quarterly", "half_yearly", "annually"}
```

Comment:
Careful, here you're using a set which is not necessarily the best option. A dictionary might be more adapted. Also, I would change the scope of this variable and put it as a class variable.